Mathlib/Analysis/Fourier/FourierTransformDeriv.lean

```diff
@@ -371,6 +683,48 @@ theorem iteratedFDeriv_fourierIntegral {N : ℕ∞}
     iteratedFDeriv ℝ n (𝓕 f) = 𝓕 (fun v ↦ fourierPowSMulRight (innerSL ℝ) f v n) :=
   VectorFourier.iteratedFDeriv_fourierIntegral (innerSL ℝ) hf h'f hn
 
+/-- The Fourier integral of the `n`-th derivative of a function is obtained by multiplying the
+Fourier integral of the original function by `(2πI L w ⬝ )^n`. -/
+theorem fourierIntegral_iteratedFDeriv {N : ℕ∞} (hf : ContDiff ℝ N f)
+    (h'f : ∀ (n : ℕ), n ≤ N → Integrable (iteratedFDeriv ℝ n f)) {n : ℕ} (hn : n ≤ N) :
+    𝓕 (iteratedFDeriv ℝ n f)
+      = (fun w ↦ fourierPowSMulRight (-innerSL ℝ) (𝓕 f) w n) := by
+  rw [← innerSL_real_flip V]
+  exact VectorFourier.fourierIntegral_iteratedFDeriv (innerSL ℝ) hf h'f hn
+
+/-- One can bound `‖w‖^n * ‖D^k (𝓕 f) w‖` in terms of integrals of the derivatives of `f` (or order
+at most `n`) multiplied by powers of `v` (of order at most `k`). -/
+lemma pow_mul_norm_iteratedFDeriv_fourierIntegral_le [FiniteDimensional ℝ V]
+    {K N : ℕ∞} (hf : ContDiff ℝ N f)
+    (h'f : ∀ (k n : ℕ), k ≤ K → n ≤ N → Integrable (fun v ↦ ‖v‖^k * ‖iteratedFDeriv ℝ n f v‖))
+    {k n : ℕ} (hk : k ≤ K) (hn : n ≤ N) (w : V) :
+    ‖w‖ ^ n * ‖iteratedFDeriv ℝ k (𝓕 f) w‖ ≤ (2 * π) ^ k * (2 * k + 2) ^ n *
+      ∑ p in Finset.range (k + 1) ×ˢ Finset.range (n + 1),
+        ∫ v, ‖v‖ ^ p.1 * ‖iteratedFDeriv ℝ p.2 f v‖ := by
+  have Z := VectorFourier.pow_mul_norm_iteratedFDeriv_fourierIntegral_le (innerSL ℝ) hf h'f hk hn
+    w w
+  simp only [innerSL_apply _ w w, real_inner_self_eq_norm_sq w, _root_.abs_pow, abs_norm,
+    mul_assoc] at Z
+  rw [pow_two, mul_pow, mul_assoc] at Z
```

Comment:
This proof comes across as slightly mysterious because the type of `Z` is never stated explicitly (but the type is very long to write down so it's maybe inevitable).